apps/root/src/common/components/transaction-steps/index.tsx

```diff
@@ -485,27 +479,25 @@ const buildApproveTokenItem = ({
           isLoading={isPendingTransaction}
         >
           {!hash ? (
-            <ContainerBox gap={3}>
-              <Button
-                onClick={isPermit2Enabled ? () => onAction() : () => onAction(amount)}
-                variant="contained"
-                sx={{ flex: 1 }}
-                size="large"
-                disabled={isPendingTransaction}
-              >
-                {isPendingTransaction ? waitingForAppvText : isPermit2Enabled ? infiniteBtnText : specificBtnText}
-              </Button>
-              {isPermit2Enabled && (
-                <Button
-                  onClick={() => onAction(amount)}
-                  size="large"
-                  variant="outlined"
-                  disabled={isPendingTransaction}
-                >
-                  {specificBtnTextAsSecondary}
-                </Button>
+            <>
+              {isCurrentStep && (
+                <ContainerBox gap={3}>
+                  <Button
+                    onClick={isPermit2Enabled && useUnlimitedApproval ? () => onAction() : () => onAction(amount)}
```

Comment:
isPermit2Enabled i think comes from agg settings, shouldn't we erase that setting?

Reply:
<img width="267" alt="Screenshot 2024-11-07 at 5 50 29 pm" src="https://github.com/user-attachments/assets/5df83dc2-7331-4887-b282-b484bdf8354f">
We actually use it quite a bit